File: src/cli/output.ts
import { BrowserPlatformError } from '../core/errors.js';

export function printJson(payload: unknown): void {
  console.log(JSON.stringify(payload, null, 2));
}

export function printErrorJson(error: unknown): void {
  if (error instanceof BrowserPlatformError) {
    printJson({ ok: false, error: { code: error.code, message: error.message, details: error.details ?? null } });
    return;
  }

  const message = error instanceof Error ? error.message : 'Unknown error';
  printJson({ ok: false, error: { code: 'UNEXPECTED_ERROR', message, details: null } });
}

File: browser_platform/cli/output.py
from __future__ import annotations
import json
from ..core.errors import BrowserPlatformError


def print_json(payload: object) -> None:
    print(json.dumps(payload, indent=2, ensure_ascii=False))


def print_error_json(error: Exception) -> None:
    if isinstance(error, BrowserPlatformError):
        print_json({'ok': False, 'error': {
            'code': error.code,
            'message': error.message,
            'details': error.details,
        }})
        return
    message = str(error) if error else 'Unknown error'
    print_json({'ok': False, 'error': {
        'code': 'UNEXPECTED_ERROR',
        'message': message,
        'details': None,
    }})
